### PR TITLE
Add support for Stellar RPC

### PR DESCRIPTION
**Pull request by noleto**
### Add Stellar RPC Support

This PR tackles https://github.com/DIN-center/din-caddy-plugins/issues/184: It brings  Stellar RPC support to the DIN Caddy plugins, allowing the gateway to route and monitor Stellar RPC traffic. The implementation follows the existing handler pattern and includes full test coverage across the new Stellar-specific behavior.

### What’s Included

* **New Stellar RPC handler**
  Adds a dedicated handler at `lib/network/stellar_rpc_handler.go`.

* **Stellar JSON-RPC 2.0 compatibility**
  Implements Stellar’s JSON-RPC 2.0 request/response structure and method routing as defined in the Stellar RPC API spec => https://developers.stellar.org/docs/data/apis/rpc/api-reference/structure/json-rpc

* **Provider health checks via `getHealth`**
  Uses Stellar RPC’s native `getHealth` endpoint for liveness/readiness monitoring.

* **Ledger-first semantics (no “blocks”)**
  Stellar does not expose blocks in the same way as EVM chains. The handler maps:

  * “block operations” → **ledger operations** (via `getLedgers`)
  * “block number” → **latest ledger sequence** (via `getLatestLedger`)

* **Network validation using passphrases**
  Validates the **Stellar network passphrase** (Mainnet/Testnet) to prevent misrouting across networks.

* **Supported RPC methods**
  `getEvents`, `getFeeStats`, `getHealth`, `getLatestLedger`, `getLedgerEntries`, `getLedgers`, `getNetwork`, `getTransaction`, `getTransactions`, `getVersionInfo`, `sendTransaction`, `simulateTransaction`

### Architecture & Platform Improvements

* **Strongly-typed handlers**
  Handler kinds are now enforced via types (no more string-based identifiers).

* **Reusable “latest height” extraction**
  Refactors the “latest block/height” payload parsing so it can be reused across handlers (now compatible with Stellar’s latest ledger sequence).

* **Brotli decompression support (`content-encoding: br`)**
  Adds Brotli decoding for Stellar providers behind certain proxies/CDNs (e.g., Cloudflare) that return Brotli-compressed responses. Uses `github.com/andybalholm/brotli`.

* **Auto-generated mocks for `NetworkHandler`**
  Adds generated mocks to improve test ergonomics and keep interfaces consistent.

### How to test locally (Stellar RPC)

1. **Update `Caddyfile.private`** (public Stellar RPC providers only):

```caddyfile
stellar-mainnet-rpc {
  methods getEvents getFeeStats getHealth getLatestLedger getLedgerEntries getLedgers getNetwork getTransaction getTransactions getVersionInfo sendTransaction simulateTransaction
  providers {
    https://mainnet.sorobanrpc.com
    https://stellar-soroban-public.nodies.app
  }
  handler stellar-rpc
  chain_id "Public Global Stellar Network"
}

stellar-testnet-rpc {
  methods getEvents getFeeStats getHealth getLatestLedger getLedgerEntries getLedgers getNetwork getTransaction getTransactions getVersionInfo sendTransaction simulateTransaction
  providers {
    https://stellar.liquify.com/api=41EEWAH79Y5OCGI7/testnet
    https://soroban-rpc.testnet.stellar.gateway.fm
    https://stellar-soroban-testnet-public.nodies.app
  }
  handler stellar-rpc
  chain_id "Test SDF Network"
}
```

2. **Build + run the DIN Router**:

```bash
make docker-build && docker compose up -d
```

3. **Run Tom’s Stellar RPC smoke test script**: https://github.com/DIN-center/din-caddy-plugins/issues/184#issuecomment-3893908723

```bash
./stellar-rpc.sh http://127.0.0.1:8000/stellar-mainnet-rpc
```

4. **Expected output**

```text
Testing Stellar RPC endpoint: http://127.0.0.1:8000/stellar-mainnet-rpc

  OK   getHealth
  OK   getVersionInfo
  OK   getNetwork
  OK   getLatestLedger
  OK   getFeeStats
  OK   getEvents (method exists, response: error -32600)
  OK   getLedgerEntries
  OK   getLedgers (method exists, response: error -32600)
  OK   getTransactions (method exists, response: error -32600)
  OK   getTransaction
  OK   sendTransaction
  OK   simulateTransaction

--- Summary ---
Passed: 12
Failed: 0
All 12 Stellar RPC methods are supported.
```

### CODE REVIEW

File: lib/network/stellar_rpc_handler.go [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+	}
+
+	if len(chainID) == 0 {
+		return fmt.Errorf("empty chain ID")
+	}
+
+	// Stellar uses network passphrase as chain ID
+	// Check for known network prefixes: "Public Global Stellar Network" or "Test SDF Network"
+	// This is more lenient and allows for variations in the full passphrase
+	validPrefixes := []string{
```
Comment (natefikru) [inline lib/network/stellar_rpc_handler.go:142]: This is peculiar and definitely different from all of the other networks, just want to double check that this is the best way to handle this, and that there isn't a single chain identifier

File: lib/network/testdata/stellar_get_latest_ledger_response.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (natefikru) [inline lib/network/testdata/stellar_get_latest_ledger_response.json:1]: We can reduce count of files by having all text fixtures organized by handler

**Review by natefikru — APPROVED** [review]
Left some nitpick reviews but overall it looks solid to me, I think this PR probably could have been split into 2 tbh.

File: modules/helpers_test.go
Comment (cds-amal) [inline modules/helpers_test.go:1]: @smoyer64 , if the suffix ends `_tests.go`, would it be leaking into production?

Comment (noleto) [reply modules/helpers_test.go:1]: this was a renaming issue, I reverted the name to helpers_test.go

File: lib/network/stellar_rpc_handler.go (hunk 1 above)
Comment (noleto) [reply lib/network/stellar_rpc_handler.go:142]: Each Stellar network has a unique identifier in the format **“[Network Name] ; [Month of Creation] [Year of Creation]”** also known as the *network passphrase*. This passphrase is used to verify that a transaction signed for one network isn’t submitted to another. More details here: [https://developers.stellar.org/docs/networks#network-passphrases](https://developers.stellar.org/docs/networks#network-passphrases)

Internally, Stellar also uses a **Network ID**, which is simply the SHA-256 hash of the network passphrase.

Comment (noleto) [reply lib/network/stellar_rpc_handler.go:142]: This raises a broader question about how our current model in the DIN Router works. It was originally designed with EVM networks in mind, then extended to EVM-compatible networks, and later expanded to other types of networks. As we continue adding services that may not even be blockchain networks, we’ll need to revisit how we handle metadata, in my opinion.

File: lib/network/testdata/stellar_get_latest_ledger_response.json (hunk 2 above)
Comment (noleto) [reply lib/network/testdata/stellar_get_latest_ledger_response.json:1]: Agreed, that’s a good point. This is probably something to address when reorganizing how we test our handlers. On a side note, initially I wrote all the test cases “inline” (generated by Claude). While reviewing them, I noticed that Claude often produces tautological tests. I then generated actual real payloads and asked Claude to write tests based on those payloads. The results were much better they caught real logic issues in my code and provided solid edge-case coverage. So something I'm in favor to be generalized for all handlers

**Comment by noleto** [comment]
> I think this PR probably could have been split into 2 tbh.

I did refactor some code beyond the Stellar requirements, but aside from the `HandlerType` refactoring, everything else was directly tied to what Stellar needed. That said, I see your point  the PRs are still quite verbose, which to me suggests that our abstractions are still too large. Handlers were a good step, but we need to push things further and keep separating concerns so that adding a new handler becomes a straightforward, low-impact change in the codebase. Definitely something to discuss.

File: lib/network/beacon_handler.go [hunk 3]
```diff
@@ -36,8 +36,8 @@ func NewBeaconChainHandler(config *NetworkConfig) *BeaconChainHandler {
 }
 
 // Metadata methods for registry
-func (h *BeaconChainHandler) GetType() string {
-	return "beacon-chain"
+func (h *BeaconChainHandler) GetType() HandlerType {
```
Comment (smoyer64) [inline lib/network/beacon_handler.go:39]: We can do more of this - excellent work typing these string constants as part of adding a feature 🔥

**Review by smoyer64 — APPROVED** [review]
